modin/core/execution/client/io.py

```diff
@@ -0,0 +1,142 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""The module holds the factory which performs I/O using pandas on a Client."""
+
+from .query_compiler import ClientQueryCompiler
+from modin.core.io.io import BaseIO
+import os
+import pandas
+
+
+class ClientIO(BaseIO):
+    """Factory providing methods for performing I/O operations using a given Client as the execution engine."""
+
+    _server_conn = None
+    _data_conn = None
+    query_compiler_cls = ClientQueryCompiler
+
+    @classmethod
+    def set_server_connection(cls, conn):
+        """
+        Set the server connection for the I/O object.
+
+        Parameters
+        ----------
+        conn : Any
+            Connection object that implements various methods.
+        """
+        cls._server_conn = conn
+
+    @classmethod
+    def set_data_connection(cls, conn):
+        """
+        Set the data connection for the I/O object.
+
+        Parameters
+        ----------
+        conn : Any
+            Connection object that is implementation specific.
+        """
+        cls._data_conn = conn
+
+    @classmethod
+    def read_csv(cls, filepath_or_buffer, **kwargs):
```

Comment:
If my understanding is correct, this is a network service, that can read arbitrary file in the file system. It seems like a security issue. I think, absolute file paths must be restricted here. Only paths, relative to the user defined directories must be allowed.

Reply:
@AndreyPavlenko that's a good question to raise. This implementation doesn't directly imply a network service in my opinion (any query compiler can be used in the service model). Restrictions on what types of files used is perhaps out of scope for this PR, but something we should potentially consider in the future.

Reply:
Sure, I don't insist on implementing it in this PR. I think, some configuration capability could be added, to restrict the arbitrary file reading.